Simplify spec, tweak tarball and (s)rpm versioning

michalfabik opened the pull request:
This commit removes some globals from the spec which needed to be set by hand before release and moves the logic for determining (s)rpm version to the Makefile. Also, move the spec to the repo root.

mgrabovsky commented:
Hm, what's the purpose of the `snapshot` variable? Couldn't we just get rid of it altogether in some way?

mgrabovsky commented:
Any progress on this?

michalfabik committed:
Simplify spec, tweak tarball and (s)rpm versioning

This commit removes some globals from the spec which needed to be set by
hand before release and moves the logic for determining (s)rpm version
to the Makefile. Also, move the spec to the repo root.

michalfabik commented:
Reworked completely.

mgrabovsky commented:
/packit build

mgrabovsky reviewed (approved):
Looks pretty good to me.

mgrabovsky commented:
@michalfabik Could you please rebase on top of current master? I've fixed the failing test and made a small change in the spec file in the process.

mgrabovsky commented:
I don't get how the F33 build fails when it [went through](https://github.com/abrt/abrt-java-connector/pull/14/commits/efada768796c992b4a8d9f22f2978e1a19acba5b) just minutes ago in the other PR.

[It fails](https://copr-be.cloud.fedoraproject.org/results/packit/abrt-abrt-java-connector-13/fedora-33-x86_64/02284430-abrt-java-connector/builder-live.log.gz) on the same test as before (throwing the _original_ error message this time around) but also on some others:

```diff
--- /builddir/build/BUILD/abrt-java-connector-1.2.0/x86_64-redhat-linux-gnu/test/outputs/run_test.log	2021-06-21 10:18:49.327280852 +0000
+++ /tmp/abrt_java_connector.xq3B29x	2021-06-21 10:19:11.646494274 +0000
@@ -27,7 +27,7 @@
 	at Test.main(Test.java:513) [file:/builddir/build/BUILD/abrt-java-connector-1.2.0/x86_64-redhat-linux-gnu/test/Test.class]
 executable: /builddir/build/BUILD/abrt-java-connector-1.2.0/x86_64-redhat-linux-gnu/test/Test.class
 Caught exception java.net.UnknownHostException in method java.net.Inet4AddressImpl.lookupAllHostAddr()
-Exception in thread "main" java.net.UnknownHostException: xyzzy: Temporary failure in name resolution
+Exception in thread "main" java.net.UnknownHostException: xyzzy: Name or service not known
 	at java.base/java.net.Inet4AddressImpl.lookupAllHostAddr(Native Method) [jrt:/java.base/java/net/Inet4AddressImpl.class]
 	at java.base/java.net.InetAddress$PlatformNameService.lookupAllHostAddr(InetAddress.java:LINENO) [jrt:/java.base/java/net/InetAddress$PlatformNameService.class]
 	at java.base/java.net.InetAddress.getAddressesFromNameService(InetAddress.java:LINENO) [jrt:/java.base/java/net/InetAddress.class]
@@ -69,26 +69,6 @@
 	at Test.networkRelatedIssues(Test.java:474) [file:/builddir/build/BUILD/abrt-java-connector-1.2.0/x86_64-redhat-linux-gnu/test/Test.class]
 	at Test.main(Test.java:514) [file:/builddir/build/BUILD/abrt-java-connector-1.2.0/x86_64-redhat-linux-gnu/test/Test.class]
 executable: /builddir/build/BUILD/abrt-java-connector-1.2.0/x86_64-redhat-linux-gnu/test/Test.class
-Caught exception java.lang.NumberFormatException in method java.lang.Long.parseLong()
-Exception in thread "main" java.lang.NumberFormatException: null
-	at java.base/java.lang.Long.parseLong(Long.java:LINENO) [jrt:/java.base/java/lang/Long.class]
-	at java.base/java.lang.Long.parseLong(Long.java:LINENO) [jrt:/java.base/java/lang/Long.class]
-	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:LINENO) [jrt:/java.base/sun/net/www/protocol/http/HttpURLConnection.class]
-	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:LINENO) [jrt:/java.base/sun/net/www/protocol/http/HttpURLConnection.class]
-	at java.base/java.net.URL.openStream(URL.java:LINENO) [jrt:/java.base/java/net/URL.class]
-	at Test.readFromURL(Test.java:238) [file:/builddir/build/BUILD/abrt-java-connector-1.2.0/x86_64-redhat-linux-gnu/test/Test.class]
-	at Test.networkRelatedIssues(Test.java:475) [file:/builddir/build/BUILD/abrt-java-connector-1.2.0/x86_64-redhat-linux-gnu/test/Test.class]
-	at Test.main(Test.java:514) [file:/builddir/build/BUILD/abrt-java-connector-1.2.0/x86_64-redhat-linux-gnu/test/Test.class]
-executable: /builddir/build/BUILD/abrt-java-connector-1.2.0/x86_64-redhat-linux-gnu/test/Test.class
-Caught exception java.io.FileNotFoundException in method sun.net.www.protocol.http.HttpURLConnection.getInputStream0()
-Exception in thread "main" java.io.FileNotFoundException: http://localhost:54321/_this_does_not_exists_
-	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:LINENO) [jrt:/java.base/sun/net/www/protocol/http/HttpURLConnection.class]
-	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:LINENO) [jrt:/java.base/sun/net/www/protocol/http/HttpURLConnection.class]
-	at java.base/java.net.URL.openStream(URL.java:LINENO) [jrt:/java.base/java/net/URL.class]
-	at Test.readFromURL(Test.java:238) [file:/builddir/build/BUILD/abrt-java-connector-1.2.0/x86_64-redhat-linux-gnu/test/Test.class]
-	at Test.networkRelatedIssues(Test.java:475) [file:/builddir/build/BUILD/abrt-java-connector-1.2.0/x86_64-redhat-linux-gnu/test/Test.class]
-	at Test.main(Test.java:514) [file:/builddir/build/BUILD/abrt-java-connector-1.2.0/x86_64-redhat-linux-gnu/test/Test.class]
-executable: /builddir/build/BUILD/abrt-java-connector-1.2.0/x86_64-redhat-linux-gnu/test/Test.class
 Caught exception java.net.MalformedURLException in method java.net.URL.<init>()
 Exception in thread "main" java.net.MalformedURLException: no protocol: @#$%^&malformed URL@#$%^&*()
 	at java.base/java.net.URL.<init>(URL.java:LINENO) [jrt:/java.base/java/net/URL.class]
```

I don't have sufficient knowledge of Java to make anything out of this.

mgrabovsky commented:
This PR looks fine to me so let's merge this and deal with the tests separately.
